cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/common/TransactionsUtility.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.etl.common;
+
+import com.google.common.base.Objects;
+import com.google.common.base.Throwables;
+import org.apache.tephra.TransactionFailureException;
+
+/**
+ * Utility class to unwrap TransactionFailureException for proper handling.
+ * This is similar to co.cask.cdap.data2.transaction.Transactions, but added here to avoid dependency
+ * on cdap-data-fabric package
+ */
+public class TransactionsUtility {
```

Comment:
I started with cdap-api. But bringing in guava dependency in cdap-api breaks security-spi. There are unit tests in place [testAuthorizerClassLoaderParentUnavailableClasses()] that assert guava to not be present.